src/tts_text/phoneme_covering_set.py

```diff
@@ -0,0 +1,196 @@
+    document["all_found_example_words"] = (
+        document["da_found_example_words"] + document["en_found_example_words"]
+    )
+    document["all_unique_phonemes"] = set(
+        document["da_unique_phonemes"] | document["en_unique_phonemes"]
+    )
+    document["all_unique_phonemes_count"] = len(document["all_unique_phonemes"])
+    return document
+
+
+def make_phoneme_covering_set(cfg: DictConfig, output_path: Path) -> list[str]:
+    """Create a phoneme covering set from the sorted wiki dataset.
+
+    A phoneme covering set is a set of strings that contains at least one example
+    word for each phoneme.
+
+    Args:
+        cfg (DictConfig): The Hydra configuration object.
+
+    Returns:
+        list[str]: The phoneme covering set.
+    """
+    sorted_dataset = sort_wiki_by_phoneme_occurence(cfg=cfg)
+
+    raw_data_path = Path(cfg.dirs.data) / "raw"
+    phoneme_path = raw_data_path / cfg.phoneme_filename
+    with open(phoneme_path) as f:
+        phonemes = json.load(f)
+    all_phone_names = [entry["name"] for entry in phonemes["da"]] + [
+        entry["name"] for entry in phonemes["en"]
+    ]
+
+    covering_set = []
+    for document in tqdm(sorted_dataset):
+        document_phonemes = document["all_phonemes"]
+        new_phoneme_found = False
+        for phoneme in document_phonemes:
+            if phoneme in all_phone_names:
+                new_phoneme_found = True
+                all_phone_names.pop(all_phone_names.index(phoneme))
+        if new_phoneme_found:
+            covering_set.append(document)
+        if len(all_phone_names) == 0:
+            break
+
+    covering_set_dataset = Dataset.from_list(covering_set)
+    example_words = get_example_words(phonemes)
+
+    # Extract only the paragraphs that contain example words
+    def extract_paragraph_with_example_word(document: dict) -> dict:
+        found_paragraphs = []
+        for paragraph in re.split(
+            "\t|\n|_NEWLINE_|_START_ARTICLE_|_START_SECTION_|_START_PARAGRAPH_",
+            document["text"],
+        ):
+            if any(word in example_words["all"] for word in paragraph.split()):
+                found_paragraphs.append(paragraph)
+
+        document["extracted_text"] = " ".join(found_paragraphs)
+        return document
+
+    covering_set_dataset = covering_set_dataset.map(
+        extract_paragraph_with_example_word,
+    )
+    dataset = [document["extracted_text"] for document in covering_set_dataset]
+    dataset_path = Path(output_path) / "phoneme_covering_set.txt"
+    with dataset_path.open("w") as f:
+        f.write("\n".join(dataset))
+
+    return dataset
+
+
+def get_example_words(phonemes: dict) -> dict:
+    """Get example words from the phonemes dict.
+
+    Args:
+        phonemes (dict): The phonemes dict.
+
+    Returns:
+        dict: The example words.
+    """
+    # Get example words
+    da_phonemes = phonemes["da"]
+    en_phonemes = phonemes["en"]
+    example_words = {}
```

Comment:
{} can be both a set and a dictionary.

```suggestion
    example_words: dict[str, list[str]] = {}
```